vault/resource_auth_backend_test.go

```diff
@@ -373,9 +373,9 @@ func checkAuthMount(backend string, checker func(*api.AuthMount) error) resource
 		for serverPath, serverAuth := range auths {
 			if serverPath == backend+"/" {
 				found = true
-				if serverAuth.Type != "github" {
-					return fmt.Errorf("unexpected auth type")
-				}
+				// if serverAuth.Type != "github" {
+				// 	return fmt.Errorf("unexpected auth type")
+				// }
```

Comment:
Seems unrelated to adding tune?

Reply:
I'm using checkAuthMount to test the tune config for the LdapAuthBackend
 vault/resource_ldap_auth_backend_test.go vault/resource_ldap_auth_backend_test.go line 252-254 and line 266-268. This test fails, when checkAuthMount checks for type "github"

checkAuthMount is only used in  vault/resource_auth_backend_test.go, e.g. lines 228-230. There is already a check for the Backend type in line 223. So I think this check is not needed here.

https://github.com/hashicorp/terraform-provider-vault/pull/997 reimplements the check for the okta backend. By removing the type check it would be possible to share the implementation for all auth backends.

If this change is approved, I can push a new version, which removes these lines.